lib/backend/calculateCompletionRate.ts

```diff
@@ -2,25 +2,53 @@ import { IssueStates } from '../enums';
 
 export interface CalculateCompletionRateOptions {
   state: IssueStates;
-  children: { state: IssueStates }[];
+  children: { state: IssueStates, children: CalculateCompletionRateOptions[] }[];
+}
+
+interface GetIssueCountsResponse {
+  open: number;
+  closed: number;
+  total: number;
+  percentClosed: number;
+}
+
+/**
+ * recursively count the total, open, and closed issues, and return an object that includes the percent closed
+ * @param {CalculateCompletionRateOptions} issue
+ * @returns
+ */
+export function getIssueCounts(issue: CalculateCompletionRateOptions): GetIssueCountsResponse {
+  let open = issue.state === IssueStates.OPEN ? 1 : 0;
+  let closed = issue.state === IssueStates.CLOSED ? 1 : 0;
+  let total = 1;
+  if ((issue.children?.length ?? 0) !== 0) {
+
+    const withChildren = issue.children.reduce(({open: accOpen, closed: accClosed, total: accTotal}, child) => {
+      const { open: childrenOpen, closed: childrenClosed, total: childrenTotal } = getIssueCounts(child);
+      const total = accTotal + childrenTotal;
+      const open = accOpen + childrenOpen;
+      const closed = accClosed + childrenClosed;
+      return {
+        total,
+        open,
+        closed,
+      };
+    }, {open, closed, total});
+    open = withChildren.open;
+    closed = withChildren.closed;
+    total = withChildren.total;
+  }
```

Comment:
we can improve it.. but I didn't want to focus on this level of improvement until I had async children done. the changes in this PR are a significant improvement over previous changes, and your changes are another improvement upon that. 

I think this would have been better handled as another issue, but I will fix and add a test for duplicate children.